docs/config_to_support_additional_workspace_directories.md

```diff
@@ -0,0 +1,101 @@
+## Config to support additional workspace directories
+
+### Issue
+https://github.com/runfinch/finch/issues/107
+
+### Approach
+
+Lima yaml file has Mounts field to handle the mount points. Similar to other Finch config like cpu and memory, it is straightforward to write additional_workspace_directories to Lima’s override.yaml file.
+
+For example, for Finch config:
+```
+cpus: 2
+memory: 4GiB
+additional_workspace_directories:
+- /Volumes/mydir
+```
+
+Finch config applier will make override.yaml like this:
+
+```
+images: []
+cpus: 2
+memory: 4GiB
+mounts:
+- location: /Volumes/mydir
+mountPoint: /Volumes/mydir
+writable: true
+sshfs:
+cache: true
+followSymlinks: false
+sftpDriver: ""
+9p:
+securityModel: none
+protocolVersion: 9p2000.L
+msize: 128KiB
+cache: mmap
+networks:
+- lima: finch-shared
+```
+
+Different to cpu and memory, the “mounts” field in override.yaml will be appended to the default mounts instead of replacing it. So we don’t have to add the default home directory to the override.yaml file. Reference (https://github.com/lima-vm/lima/blob/master/pkg/limayaml/defaults.go#L410)
+
+### Error handling consideration
+
+There are lots of error cases if users directly pass random locations to Lima. For example, when users try to set “/” as in additional workspace directories.
+
+*Option 1 Directly pass the Lima and return whatever error Lima returns. Contribute to Lima to enhance error handling if needed. (Recommended)*
+
+Pros:
+
+* Easy to implement
+
+Cons:
+
+* Sometimes the error log in Lima context may not be descriptive in Finch context
```

Comment:
Another option is to wrap Lima error into a more descriptive finch error

Reply:
Can we catch Lima errors and modify it for Finch ?

Reply:
We used buffer to do a simple replacement in nerdctl [command](https://github.com/runfinch/finch/blob/d8174ff773f0f92ec94d6d97c753a872a98f74a0/cmd/finch/nerdctl.go#L76). However it will become very tricky when doing complicated replacement. It may mess up the messages when there is overlap or there is a message change in upstream. It would also make any commands that require stdin exit. I tend to not do message replacing unless there is no other alternative.